prody/proteins/pdbfile.py

```diff
@@ -590,16 +593,34 @@ def _parsePDBLines(atomgroup, lines, split, model, chain, subset,
             resnames[acount] = resname
             chainids[acount] = chid
             if isPDB:
-                try:
-                    resnums[acount] = int(line[22:26])
-                except ValueError:
+                if dec:
+                    try:
+                        resnum = int(line[22:26])
+                    except ValueError:
+                        dec = False
+
+                    if resnums[acount-2] >= MAX_N_RES:
+                        dec = False
+                else:
                     try:
-                        resnums[acount] = int(line[22:26], 16)
+                        resnum = int(line[22:26], 16)
+                        
                     except ValueError:
                         LOGGER.warn('failed to parse residue number in line {0}. Assigning it by incrementing.'
                                     .format(i))
-                        resnums[acount] = resnums[acount-1]+1
-                icodes[acount] = line[26] 
+                        resnum = resnums[acount-1]+1
+
+                icode = line[26] 
+
+                if icode.isnumeric() and dec:
+                    if not warned_5_digit:
+                        LOGGER.warn('parsed 5 digit residue number including numeric insertion code')
+                        warned_5_digit = True
+                    resnum = int(resnum + icode)
```

Comment:
`resnum` at this point seems to be an `int` so this might not work as intended. Please double check.

Reply:
This should now be fixed.

Reply:
Actually, that's not the problem. The problem is that it now expects it to be hex.

Reply:
Ok sorted in the coming commit.